From bb596ecb5a28d0cb3fa01491652dd4d0e4f9d33b Mon Sep 17 00:00:00 2001
From: Rui Loura <rui@joyent.com>
Date: Mon, 30 Apr 2018 14:07:30 +0000
Subject: [PATCH] TRITON-612 fw-agent should be aware of alternate admin
 nictags Reviewed by: Cody Peter Mello <cody.mello@joyent.com> Approved by:
 Cody Peter Mello <cody.mello@joyent.com>

---
 lib/config.js | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/lib/config.js b/lib/config.js
index c01d4a8..7709446 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -25,11 +25,16 @@ var VError = require('verror').VError;
 function findSysinfoAdminIP(si) {
     var iface;
     var ifaces = si['Network Interfaces'];
+    var tag = 'admin';
+
+    if (si['Admin NIC Tag']) {
+        tag = si['Admin NIC Tag'];
+    }
 
     for (var i in ifaces) {
         iface = ifaces[i];
 
-        if (iface['NIC Names'] && iface['NIC Names'].indexOf('admin') !== -1) {
+        if (iface['NIC Names'] && iface['NIC Names'].indexOf(tag) !== -1) {
             return iface.ip4addr;
         }
     }
-- 
2.21.0

